assets/javascripts/discourse/controllers/admin-plugins-explorer.js.es6

```diff
@@ -117,16 +127,41 @@ export default Ember.Controller.extend({
       .finally(() => this.set("loading", false));
   },
 
+  _readQueryFile(file) {
+    const reader = new FileReader();
+    reader.onload = (evt) => {
```

Comment:
Probably want to implement `FileReader.onerror` here as well to be able to reset the state and show a message on error. If this is a pattern we use in a few places in future we may want to refactor this out into a `lib` that we can import. No need for that yet though as this is the only place we are doing this so far :)

Reply:
E.g. `app/lib/file-reader.js`